src/actions/actions.ts

```diff
@@ -610,6 +610,18 @@ class CommandFormatCode extends BaseCommand {
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
     await vscode.commands.executeCommand("editor.action.format");
+
+    let selection = vscode.window.activeTextEditor.selection;
+    let line = selection.start.line;
+
+    if (selection.start.isAfter(selection.end)) {
+      line = selection.end.line;
+    }
+
+    let newCursorPosition = new Position(line, 0);
+    vimState.cursorPosition = newCursorPosition;
+    vimState.cursorStartPosition = newCursorPosition;
+    vscode.window.activeTextEditor.selection = new vscode.Selection(newCursorPosition, newCursorPosition);
```

Comment:
You want to avoid "updating the view" (making any visible changes) inside an Action in actions.ts. In this case it's easy because the state machine will automatically update the selection to start at cursorStartPosition and end at cursorPosition anyways.